### ISSUE TITLE
Noticeable errors in results

### ISSUE THREAD

**Issue by yuanj07** [manycore-research/SpatialLM#44]
Thanks for sharing such an interesting work. I tried the code using my own data. Point clouds are created using dust3r and rotated to align z axis with gravity direction. The true scale is also estimated. It seems that objects like beds and walls are detected well, while other objects are often problematic (incorrect labels and bounding boxes). The attached images show one example. The dinning table is labeled as combination sofa, kitchen island as dining table, glass door as window. Are these errors expected, or is there any preprocessing step I should apply to point clouds? 

![Image](https://github.com/user-attachments/assets/524faabe-a751-4243-aaf5-3a84fc8a4bbb)
![Image](https://github.com/user-attachments/assets/c4cefd05-1405-490e-83ea-5f9a62b94868)

**Comment by alan0717-cn** [comment]
Hello, I would like to know how you align the coordinates to the z-axis. If it is convenient for you, please tell me.

**Comment by yunho-c** [comment]
From reading previous issues, it seems that accuracy is primarily limited by narrow training data distributions (which only consists of single-room indoor environments in residential apartments; primarily synthetic data). Fine-tuning seems necessary to improve accuracy especially when they do not closely match the conditions of training data.

**Comment by SamMaoYS** [comment]
Hi, thank you for your valuable feedback.

I think the error is likely due to differences in data distribution between this test case and our training data. Our dataset mainly features Chinese apartment floorplans, where kitchen islands are rare and rooms like the kitchen and living room are usually separated. In contrast, "open kitchen" layouts with kitchen islands are more common in other regions, where the kitchen and living room share the same undivided space.

As a result, the model does not have a strong understanding of kitchen islands and tends to misclassifies kitchen islands as dining tables. It then infers that two dining tables in one room are unlikely, and assumes that the actual dining table is more likely part of a living room setup, leading to mistakenly labels the actual dining area as a sofa-coffee table combination.

Improving the quality of point clouds, by removing floating points and isolated areas, may also help. This can be achieved by increasing the confidence threshold in DUSt3R or applying point cloud cleaning techniques.

We appreciate your interest and feedback, this is our first attempt at applying LLMs to 3D room layout understanding, and there’s definitely room for improvement. We're continuing to refine the model with more diverse data, especially incorporating a wider variety of layouts and architectural styles, as well as better training strategies, and we hope future versions will work better on your use cases.

**Comment by yuanj07** [comment]
Thank you for your reply. 

To answer the question above, based on surface normals and up vectors in estimated camera pose, the vertical direction can be reasonably estimated.
